indico/modules/events/editing/controllers/backend/editable_list.py

```diff
@@ -0,0 +1,57 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2020 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+from flask import jsonify, request
+
+from indico.modules.events import Event
+from indico.modules.events.contributions.models.contributions import Contribution
+from indico.modules.events.editing.controllers.base import RHEditingBase
+from indico.modules.events.editing.models.editable import EditableType
+
+
+class RHEditableList(RHEditingBase):
+    """Return editable list of the event for the editable type."""
+    def _process_args(self):
+        RHEditingBase._process_args(self)
+        self.event = Event.get(int(request.view_args['confId']))
+        self.editable_type = EditableType[request.view_args['type']]
+        self.contributions = Contribution.query.with_parent(self.event)
+        self.is_code_present = False
+        self.serialized_contributions = []
+        for contribution in self.contributions:
+            serialized_editable = None
+            if contribution.editables:
+                for editable in contribution.editables:
+                    if editable.type.name == self.editable_type.name:
+                        serialized_editable = self.serialize_editable(editable)
+            self.serialized_contributions.append(self.serialize_contribution(contribution, serialized_editable))
```

Comment:
I am not sure if that is the best place to serialize the contributions. Usually we "deserialize" the request arguments to a Pythonic form in `_process_args` which then can be serialized inside RH (what you are doing with `self.contributions`). Maybe to keep the `process_args` small and have only one responsibility you should move the serialization to a separate method which then would be called from `_process`.

Reply:
indeed, this is way too much logic for `_process_args` - this all belongs in `_process`. also, i think we could make use of a marshmallow schema

Reply:
Do we need schema here?

Reply:
I think using one would certainly be cleaner.